lib/puppet/indirector/facts/rest.rb

```diff
@@ -7,6 +7,26 @@ class Puppet::Node::Facts::Rest < Puppet::Indirector::REST
   def save(request)
     raise ArgumentError, _("PUT does not accept options") unless request.options.empty?
 
+    return legacy_save(request) unless use_http_client?
+
+    session = Puppet.lookup(:http_session)
+    api = session.route_to(:puppet)
+    api.put_facts(
+      request.key,
+      facts: request.instance,
+      environment: request.environment.to_s
```

Comment:
Do we have tests validating that this is a string? We should double check that

Reply:
Updated

Reply:
Also added commit description